### DISCUSSION TITLE
Using spack in dark site

### DISCUSSION THREAD

**Discussion by asfbar** [spack/spack#34304] [General]
Hello,

Can i install and use spack in a dark site (rocky linux server without an internet connection)?
i succeed to install spack and also create a local mirror with all the apps that i want but i see that when spack 
installing SW (in normal environment with internet) it also get files from repos like github,ftp etc.

Thank you
Assaf.

**Comment by tgamblin** [comment]
You can -- you will need to create a mirror of all the sources you might fetch, and for the things Spack needs to bootstrap.

For bootstrapping, see the `spack bootstrap mirror` command in [Creating a mirror for air-gapped systems](https://spack.readthedocs.io/en/latest/bootstrapping.html#creating-a-mirror-for-air-gapped-systems).

For creating your own mirror, take a look at [`spack mirror create`](https://spack.readthedocs.io/en/latest/mirrors.html#spack-mirror-create). It sounds like you tried that already, but you likely need to fetch sources for all your dependencies with `--dependencies`, and you may also want to fetch sources/resources/patches for all versions of what you're installing, to make it easier to change things once you're on the closed machine.  You can do that with `--all`.
